### PR TITLE
GeometryTestUtils should always generate valid polygons

### LINKED ISSUE #58032: [CI] Seed-specific failure of TriangleTreeTests.testRandomPolygonIntersection in 7.x
On the 7.x branch `TriangleTreeTests.testRandomPolygonIntersection` fails reproducibly using:

```
./gradlew ':x-pack:plugin:spatial:test' --tests "org.elasticsearch.xpack.spatial.index.fielddata.TriangleTreeTests.testRandomPolygonIntersection" -Dtests.seed=FE54FE32BC33F7CF -Dtests.security.manager=true -Dtests.locale=sk-SK -Dtests.timezone=America/Santiago -Druntime.java=8
```

The error is:

```
org.locationtech.spatial4j.exception.InvalidShapeException: Cannot determine orientation: edges adjacent to (-5.2298701235708654E-183,0.0) coincide
```

Changing the seed makes it pass, so it's a seed-specific failure.

I saw this in a PR build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-2/1186/testReport/junit/org.elasticsearch.xpack.spatial.index.fielddata/TriangleTreeTests/testRandomPolygonIntersection/

### PR DESCRIPTION
**Pull request by iverase**
Make sure we always generate legal polygons, e.g they have area.

fixes #58032

### CODE REVIEW

**Comment by iverase** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by iverase** [comment]
@elasticmachine run elasticsearch-ci/2

File: test/framework/src/main/java/org/elasticsearch/geo/GeometryTestUtils.java [hunk 1]
```diff
@@ -97,6 +97,9 @@ public static Point randomPoint(boolean hasAlt) {
 
     public static Polygon randomPolygon(boolean hasAlt) {
         org.apache.lucene.geo.Polygon lucenePolygon = GeoTestUtil.nextPolygon();
+        while(area(lucenePolygon) == 0) {
```
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/geo/GeometryTestUtils.java:100]: We often use `randomValueOtherThanMany(p -> area(p) == 0, GeoTestUtil::nextPolygon);` to do this kind of thing. It isn't faster or better than what you have, but it is more common.

**Review by nik9000 — APPROVED** [review]

**Comment by iverase** [comment]
@elasticmachine run elasticsearch-ci/1 
@elasticmachine run elasticsearch-ci/default-distro

**Review by imotov — APPROVED** [review]
Makes sense! Thanks!

**Comment by iverase** [comment]
@elasticmachine update branch

**Pull request merged**